src/mito2/src/error.rs

```diff
@@ -754,6 +755,23 @@ pub enum Error {
         source: Arc<Error>,
     },
 
+    #[snafu(display("Internal error occurred in remote job scheduler: {}", reason))]
+    RemoteJobScheduler {
+        #[snafu(implicit)]
+        location: Location,
+        reason: String,
+        // Keep the waiters in the error so that we can notify them when fallback to the local compaction.
+        waiters: Vec<OutputTx>,
+    },
```

Comment:
Maybe we should define a dedicated error struct for this. This avoids the remote scheduler returns other error variants.
https://docs.rs/snafu/latest/snafu/index.html#struct-style

Reply:
I add the error `RemoteJobSchedulerError` that used for remote job scheduler.